### PR TITLE
[Smoke Test][rocRAND] Ported over rtest.py from rocPRIM and added smoke test option

### PR DESCRIPTION
**Pull request by NguyenNhuDi**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,14 @@
 Documentation for rocRAND is available at
 [https://rocm.docs.amd.com/projects/rocRAND/en/latest/](https://rocm.docs.amd.com/projects/rocRAND/en/latest/)
 
+
+## (Unreleased) rocRAND-3.2.0 for ROCm 6.4
```
Comment (amd-jnovotny) [inline CHANGELOG.md]:
```suggestion
## (Unreleased) rocRAND 3.2.0 for ROCm 6.4
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,14 @@
...
+
+### Added
+* rtest.py, a python script to run unit tests
+  * added a smoke test option. Smoke tests can now be ran with python3 rtest.py -t/--test smoke
```
Comment (amd-jnovotny) [inline CHANGELOG.md]:
```suggestion
  * added a smoke test option. To run the smoke tests, use `python3 rtest.py -t/--test smoke`.
```

**Review by amd-jnovotny — APPROVED** [review]
A couple of small changes for the changelog. Docs look good otherwise.

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,17 @@
 Documentation for rocRAND is available at
 [https://rocm.docs.amd.com/projects/rocRAND/en/latest/](https://rocm.docs.amd.com/projects/rocRAND/en/latest/)
 
+
+## (Unreleased) rocRAND 3.2.0 for ROCm 6.4
+
+### Added
+* Added smoke test options, which runs a subset of the unit tests and ensuring that less than 2gb of VRAM will be used
+  * Smoke tests can be ran with `[--emulation|-e|--test|-t]=smoke`
+* Added `--emulation` option added for `rtest.py`
```
Comment (amd-jnovotny) [inline CHANGELOG.md]:
```suggestion
### Added

* Added smoke test options, which runs a subset of the unit tests and ensures that less than 2gb of VRAM will be used
  * Smoke tests can be run using `[--emulation|-e|--test|-t]=smoke`
* Added `--emulation` option for `rtest.py`
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,17 @@
...
+
+### Changed
+* `--test|-t` no longer a required flag for `rtest.py`, instead user can use either `--emulation|-e` or `--test|-t` but not both
```
Comment (amd-jnovotny) [inline CHANGELOG.md]:
```suggestion
### Changed

* `--test|-t` is no longer a required flag for `rtest.py`. Instead, the user can use either `--emulation|-e` or `--test|-t`, but not both.
```

**Review by amd-jnovotny — APPROVED** [review]
Thank you for updating the changelog using the proper format. I just have a few editorial suggestions for you. Otherwise, the changelog looks good from a docs perspective.

**Review by umfranzw — APPROVED** [review]
The code looks good to me. The CI build failures look unrelated to this change.

**Pull request merged**